### PR TITLE
[WIP] Run examples in our CI

### LINKED ISSUE #93: Check examples in CI
We need to check the examples during CI. Otherwise, we may break them without realizing it and since they are the first thing that a new user will look at, it will give a poor impression of DTK.

### PR DESCRIPTION
**Pull request by dalg24**
Fixes #93

- [x] Configure with `DataTransferKit_ENABLE_EXAMPLES` turned `ON`.
- [x] Use `TRIBITS_ADD_EXECUTABLE_AND_TEST` macro in place of `TRIBITS_ADD_EXECUTABLE` to register examples.

### CODE REVIEW

**Comment by sslattery** [comment]
We are going to have to update the docker image to configure libmesh with netcdf if we want to run the examples with exodus files.

**Comment by dalg24** [comment]
Libmesh was configured with `--disable-netcdf` [See our Dockerfile](https://github.com/ORNL-CEES/DataTransferKit/blob/master/docker/Dockerfile_stack#L158).  We can produce a new image with an updated Trilinos and Libmesh with NetCDF support.

**Comment by dalg24** [comment]
OK tests are failing because an exception is raised in `docker_run.py`.  See [log on Jenkins](https://jupyterdocker.ornl.gov:8080/job/DataTransferKit-experimental/172/console).  It looks like a time out.  Looking more into Docker Python client API, there is a [`timeout` argument](https://docker-py.readthedocs.io/en/stable/api/#client-api) when we [instantiate the `Client` class](https://github.com/ORNL-CEES/Cap/blob/master/docker/jenkins/docker-run.py#L17).  Not sure what the default value is but I suppose were can try to increase it.

**Comment by dalg24** [comment]
Digging more into this, I found the default timeout to be 60 seconds. (See [here](https://github.com/docker/docker-py/blob/d56b2d3dc87b3d12fad197fa2cd19f3ea3f7e5d1/docker/api/client.py#L79) and [here](https://github.com/docker/docker-py/blob/d56b2d3dc87b3d12fad197fa2cd19f3ea3f7e5d1/docker/constants.py#L6).)

**Comment by sslattery** [comment]
OK - some of the examples will take longer in debug

**Comment by dalg24** [comment]
For reference, in ORNL-CEES/Cap#243 I updated the docker-run.py script to be able to relax `timeout`.  I am setting it up to 5min.

**Comment by dalg24** [comment]
OK, it did work.  See [Jenkins build #182](https://jupyterdocker.ornl.gov:8080/job/DataTransferKit-experimental/182).  The log is insanely big, nothing gets lost, all the configuration information is there.  We did not hit the timeout.
Some examples are failing though and the code format is not respected.

**Comment by dalg24** [comment]
Rebased and forced the push

**Comment by dalg24** [comment]
What do we do about the libMesh examples that are failing?

**Comment by aprokop** [comment]
@dalg24 Have those examples work in the past? If yes, fix them.

**Comment by dalg24** [comment]
At one point they did, but they never were part of the test regression and we had never run them within the container.

**Comment by dalg24** [comment]
I did another forced push.  Fixed the code format and squashed.  Sorry...

**Comment by sslattery** [comment]
i need to take a look at those examples

**Comment by dalg24** [comment]
Jenkins build #185
```
Test  #37: DataTransferKitOperators_Integral_test_MPI_1 ........................   Passed  255.37 sec
```
versus #186
```
Test  #37: DataTransferKitOperators_Integral_test_MPI_1 ........................   Passed   69.62 sec
```

File: packages/Adapters/Moab/example/interpolation.cpp [hunk 1]
```diff
@@ -338,7 +344,19 @@ int main( int argc, char *argv[] )
 
     error_l2_norm = std::sqrt( error_l2_norm );
     tag_l2_norm = std::sqrt( tag_l2_norm );
-    std::cout << "|e|_2 / |f|_2: " << error_l2_norm / tag_l2_norm << std::endl;
+    double pass_criteria = error_l2_norm / tag_l2_norm;
+    std::cout << "|e|_2 / |f|_2: " << pass_criteria << std::endl;
+
+    std::cout << std::endl;
+    std::cout << "End Result: TEST ";
+    if ( pass_criteria < 1.0e-2 )
```
Comment (dalg24) [inline packages/Adapters/Moab/example/interpolation.cpp]: That tolerance looks super relaxed to me.  Why is that so high?

Comment (sslattery) [inline packages/Adapters/Moab/example/interpolation.cpp]: The test function has absolute values in it so i suppose the first order discretization isn't doing as good of a job compared to the analytic result?

**Comment by aprokop** [comment]
@sslattery Is 4de571e really part of this PR? I would say it goes beyond the scope of this PR, and you could simply set `OMP_NUM_THREADS=1` in the scripts to run right now.

**Comment by sslattery** [comment]
I disagree with setting `OMP_NUM_THREADS` to 1. We want the kokkos tests to run with multiple threads. In addition, DTK 2 code is not intended to be threaded so the correct code is to template on a serial node. 

Finally, I was ultimately motivated to make this change in this PR because it reduced test runtimes by 25% and we have decided to use this PR to reduce runtimes of tests. The fact that the code is now correct was a nice side effect of that.

**Comment by sslattery** [comment]
@sslattery will reduce mesh sizes in example and then will cherry-pick node template commit with @dalg24 into a new PR

**Comment by sslattery** [comment]
Now seeing 6x reduction in test run time with smaller meshes and serial node refactor. Going to look at ETI types being instantiated to see if any are not needed.

**Comment by sslattery** [comment]
OK it looks like the tpetra ETI types are OK but DTK is instantiating the scalar types with integers. I'm surprised these weren't turned off by the configure but they should be removed.

**Comment by aprokop** [comment]
@sslattery The ETI code you ported to DTK should instantiate it on the same scalar types as Tpetra, right? Tpetra usually instantiates integer scalars due to some partitioning code using integer multivectors for migration.

**Comment by sslattery** [comment]
right @aprokop be we added the loop from ifpack2 that should strip the scalar types. This must not be working correctly.

**Comment by dalg24** [comment]
@sslattery dropped the ETI changes from this PR.  Merging as is.

**Pull request merged**